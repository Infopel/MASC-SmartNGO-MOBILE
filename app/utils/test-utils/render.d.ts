export type RenderOptions = {
  wrapper?: React.ComponentType<any>
  createNodeMock?: (element: React.ReactElement) => any
}
export type AllTheProvidersProps = { children: React.ReactNode }
