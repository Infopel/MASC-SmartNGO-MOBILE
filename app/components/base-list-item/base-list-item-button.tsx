import { useIsPressed } from "components/hooks/button-hooks"
import { IPressableProps, Pressable, usePropsResolution, useToken } from "native-base"
import React from "react"
interface BaseListItemButtonProps extends Omit<IPressableProps, "onPressOut" | "onPressIn"> {


}
export function BaseListItemButton({ children,...rest }: BaseListItemButtonProps) {

    const { isPressed, pressableProps } = useIsPressed()
    const { ...btnProps } = usePropsResolution('button', { variant: 'ghost', ...rest }, { isPressed })

    const color = useToken('colors', 'primary.500')

    return (
        <Pressable android_ripple={{ color }} {...btnProps}  {...pressableProps} p='2'>
            {children}
        </Pressable>
    )
}