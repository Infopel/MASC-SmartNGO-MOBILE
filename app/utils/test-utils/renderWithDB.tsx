jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  