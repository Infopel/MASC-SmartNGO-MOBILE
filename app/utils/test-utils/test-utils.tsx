import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

