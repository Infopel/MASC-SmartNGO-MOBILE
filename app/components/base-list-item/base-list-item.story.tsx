import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { BaseListItem } from "./base-list-item"

storiesOf("BaseListItem", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <BaseListItem m='2' />
      </UseCase>
    </Story>
  ))
