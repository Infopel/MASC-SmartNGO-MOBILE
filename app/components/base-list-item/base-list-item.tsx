import { Card } from "components/card/card"
import { Box, IStackProps, Pressable, useToken } from "native-base"
import * as React from "react"
import { BaseListItemButton } from "./base-list-item-button"

export interface BaseListItemProps extends IStackProps {
  onPress?: () => void
  accessibilityLabel?: string
}

/**
 * Describe your component here
 */
export function BaseListItem({ children, onPress, accessibilityLabel, ...rest }: BaseListItemProps) {

  return (
    <Card borderLeftWidth={6} borderLeftColor='primary.500' {...rest}>
      {onPress ? <BaseListItemButton onPress={onPress} accessibilityLabel={accessibilityLabel} children={children} /> : <Box>{children}</Box>}
    </Card>
  )
}
