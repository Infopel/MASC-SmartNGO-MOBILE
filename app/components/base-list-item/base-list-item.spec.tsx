import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { BaseListItem } from "./base-list-item"



describe("<BaseListItem>",()=>{
  it("render", async function () {
    const component = <BaseListItem></BaseListItem>
  
    expect(render(component).toJSON()).toMatchSnapshot()
  })
})
