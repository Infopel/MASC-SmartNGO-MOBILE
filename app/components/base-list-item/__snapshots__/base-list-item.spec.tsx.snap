// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseListItem> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "backgroundColor": "#fafaf9",
      "borderBottomColor": "#fafafa",
      "borderLeftColor": "#ef8812",
      "borderLeftWidth": 6,
      "borderRadius": 6,
      "borderRightColor": "#fafafa",
      "borderTopColor": "#fafafa",
      "borderWidth": 1,
      "elevation": 2,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    dataSet={Object {}}
    style={Object {}}
  />
</View>
`;
